src/btree/bt_split.c

```diff
@@ -1853,6 +1855,8 @@ __split_insert(WT_SESSION_IMPL *session, WT_REF *ref)
     /* The second page in the split is a new WT_REF/page pair. */
     WT_ERR(__wt_page_alloc(session, type, 0, false, &right));
 
+    right->block_meta.page_id = WT_BLOCK_INVALID_PAGE_ID;
```

Comment:
I see that you set the page ID to `WT_BLOCK_INVALID_PAGE_ID` here and in several other places. Why is that? The page can't be written with an invalid ID.

Reply:
This means the page is either deleted or a new page. If the page is reconciled again, we must write a full page.